test_runner/src/test/kotlin/ftl/util/UtilsTest.kt

```diff
@@ -172,11 +169,16 @@ class UtilsTest {
                 )
             )
         }
+
+        // will
+        exit.expectSystemExitWithStatus(1)
+        exit.checkAssertionAfterwards {
+            assertTrue(output.log.contains("1"))
+            assertTrue(output.log.contains("Error: Matrix failed: 2"))
+        }
+
         // when
         withGlobalExceptionHandling(block)
-        // then
-        assertTrue(output.log.contains("Error: Matrix failed: 1"))
-        assertTrue(output.log.contains("Error: Matrix failed: 2"))
```

Comment:
this test will be updated cause assert methods was not called after `System.exit`, so I introduced new testing behavior given-will-when 😄 to make them work